ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/merge/helpers/MergeTransitionHelpers.java

```diff
@@ -97,4 +98,32 @@ public MergeTransitionHelpers(SpecConfigMerge specConfig) {
   private boolean isBelowTotalDifficulty(final PowBlock powBlock) {
     return powBlock.getTotalDifficulty().compareTo(specConfig.getTerminalTotalDifficulty()) < 0;
   }
+
+  /* block proposal helpers */
+
+  public SafeFuture<Optional<PowBlock>> getTerminalPowBlock(
+      final ExecutionEngineChannel executionEngine) {
+    if (!specConfig.getTerminalBlockHash().isZero()) {
+      return executionEngine.getPowBlock(specConfig.getTerminalBlockHash());
+    } else {
+      return SafeFuture.completedFuture(
+          getPowBlockAtTerminalTotalDifficulty(
+              executionEngine.getPowChainHead().join(),
+              specConfig.getTerminalTotalDifficulty(),
+              executionEngine));
+    }
+  }
+
+  private Optional<PowBlock> getPowBlockAtTerminalTotalDifficulty(
+      final PowBlock head,
+      final UInt256 totalDifficulty,
+      final ExecutionEngineChannel executionEngine) {
+    PowBlock block = head;
+    Optional<PowBlock> parent = Optional.empty();
+    while (block.getTotalDifficulty().compareTo(totalDifficulty) >= 0) {
+      parent = Optional.of(block);
+      block = executionEngine.getPowBlock(block.getParentHash()).join().orElseThrow();
```

Comment:
We'd avoid using join here too.  Fortunately I don't think this loop is necessary because the EL will refuse to import blocks if their parent has already reached TTD.  So we can just get the head block and check it exceeds TTD without needing to walk back through the chain.